assets/js/modules/analytics-4/components/dashboard/EnhancedMeasurementActivationBanner/index.js

```diff
@@ -0,0 +1,172 @@
+	isValidWebDataStreamID,
+} from '../../../utils/validation';
+
+const { useSelect, useDispatch } = Data;
+
+function EnhancedMeasurementActivationBanner() {
+	const [ step, setStep ] = useState( ACTIVATION_STEP_SETUP );
+	const [
+		isEnhancedMeasurementInitiallyDisabled,
+		setIsEnhancedMeasurementInitiallyDisabled,
+	] = useState( undefined );
+
+	const propertyID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getPropertyID()
+	);
+
+	const webDataStreamID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getWebDataStreamID()
+	);
+
+	const isBannerDismissed = useSelect( ( select ) =>
+		select( CORE_USER ).isItemDismissed(
+			ENHANCED_MEASUREMENT_ACTIVATION_BANNER_DISMISSED_ITEM_KEY
+		)
+	);
+
+	const hasAnalytics4Access = useSelect( ( select ) =>
+		select( CORE_MODULES ).hasModuleOwnershipOrAccess( 'analytics-4' )
+	);
+
+	const isEnhancedMeasurementStreamEnabled = useSelect( ( select ) => {
+		if (
+			! (
+				isValidPropertyID( propertyID ) &&
+				isValidWebDataStreamID( webDataStreamID ) &&
+				hasAnalytics4Access
+			)
+		) {
+			return undefined;
+		}
+
+		return select( MODULES_ANALYTICS_4 ).isEnhancedMeasurementStreamEnabled(
+			propertyID,
+			webDataStreamID
+		);
+	} );
+
+	const { dismissItem } = useDispatch( CORE_USER );
+
+	const { isTooltipVisible } = useTooltipState(
+		ENHANCED_MEASUREMENT_ACTIVATION_BANNER_TOOLTIP_STATE_KEY
+	);
+
+	const showTooltip = useShowTooltip(
+		ENHANCED_MEASUREMENT_ACTIVATION_BANNER_TOOLTIP_STATE_KEY
+	);
+
+	function handleDismiss() {
+		showTooltip();
+		dismissItem(
+			ENHANCED_MEASUREMENT_ACTIVATION_BANNER_DISMISSED_ITEM_KEY,
+			{
+				expiresInSeconds: getTimeInSeconds( 'month' ),
+			}
+		);
+	}
+
+	const handleSubmit = useCallback( () => {
+		if ( step === ACTIVATION_STEP_SETUP ) {
+			setStep( ACTIVATION_STEP_SUCCESS );
+		}
+	}, [ step ] );
+
+	useEffect( () => {
+		if (
+			isEnhancedMeasurementStreamEnabled === false &&
+			isEnhancedMeasurementInitiallyDisabled === undefined
+		) {
+			setIsEnhancedMeasurementInitiallyDisabled( true );
+		}
+	}, [
+		isEnhancedMeasurementInitiallyDisabled,
+		isEnhancedMeasurementStreamEnabled,
+	] );
+
+	if ( isTooltipVisible ) {
+		return (
+			<AdminMenuTooltip
+				title={ __(
+					'Enable enhanced measurement later here',
+					'google-site-kit'
+				) }
+				content={ __(
+					'You can always turn on enhanced measurement later in Site Kit Settings.',
+					'google-site-kit'
+				) }
+				dismissLabel={ __( 'Got it', 'google-site-kit' ) }
+				tooltipStateKey={
+					ENHANCED_MEASUREMENT_ACTIVATION_BANNER_TOOLTIP_STATE_KEY
+				}
+			/>
+		);
+	}
+
+	if ( ! isEnhancedMeasurementInitiallyDisabled || isBannerDismissed ) {
+		return null;
+	}
+
+	switch ( step ) {
+		case ACTIVATION_STEP_SETUP:
+			return (
+				<SetupBanner
+					onSubmitSuccess={ handleSubmit }
+					onDismiss={ handleDismiss }
+				/>
+			);
+		case ACTIVATION_STEP_SUCCESS:
+			return <SuccessBanner />;
+		default:
+			return null;
+	}
```

Comment:
If the switch case that useful here? We can use less indentation and arguably a clearer, "return early" flow using if statements:

```suggestion
	if ( step === ACTIVATION_STEP_SETUP ) {
		return (
			<SetupBanner
				onSubmitSuccess={ handleSubmit }
				onDismiss={ handleDismiss }
			/>
		);
	}

	if ( step === ACTIVATION_STEP_SUCCESS ) {
		return <SuccessBanner />;
	}

	return null;
```